src/PersistentReducedStateStream.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  BehaviorSubject,
+  Observable,
+  Subscription,
+  Subject,
+  Operator,
+  Subscriber,
+  TeardownLogic,
+} from 'rxjs';
+import { ActionStream } from './types/helpers';
+import { RegisteredReducer, combineReducers } from './reducer';
+import { defaultErrorSubject } from './internal/defaultErrorSubject';
+import { markName } from './internal/markers';
+import { tag } from 'rxjs-spy/operators';
+
+/**
+ * PersistentReducedStateStream is an Observable interface for state streams.
+ *
+ * The state stream has an inner subject which produces values to its subscribers.
+ *
+ * To begin reducing values, call `.startReducing(action$)`
+ * To stop reducing, call `.stopReducing()` (this closes the inner subject)
+ *
+ * The latest value of the state stream can be read synchronously by accessing
+ * `.state` (as long as the inner subject isn't closed)
+ */
+export class PersistentReducedStateStream<State> extends Observable<State> {
+  private subject: BehaviorSubject<State>;
+  private reducerSubscription?: Subscription;
+  public name: string;
+  private reducers: RegisteredReducer<State, any>[];
+  private errorSubject: Subject<any>;
+
+  /**
+   * Subscribes to the action stream and starts reducing this state stream.
+   * startReducing does not always immediately emit a value since the value
+   * might have already been emitted when the state stream was created.
+   *
+   * initialState can be passed to reset the state of the stream to some other value.
+   * This can be useful for hot-reloading
+   *
+   * startReducing will recreate the inner subject if it was previously closed
+   */
+  startReducing = (action$: ActionStream, initialState?: State) => {
+    // Reset the subject value if it doesn't match the desired initialState
+    // This is normally used for hot-reloading
+    if (this.subject.closed && initialState) {
```

Comment:
Just found a small (unrelated) bug. This will not hot-reload correctly if `initialState` is falsy.

It should be:
```typescript
if (this.subject.closed && initialState !== undefined) {
```

Reply:
Good catch!
I try to stick to `===` and `!==`, but didn't review this properly when copying over.